### ISSUE TITLE
Schema <[object Object]> already exists with different definition

### ISSUE THREAD

**Issue by SanjeevMurthy** [tdegrunt/jsonschema#213]
Getting ``Schema <[object Object]> already exists with different definition`` error while trying to refer 2 schemas in one.

Please correct me if am doing something wrong:

> *Coupons Schema* in coupons.js
```
const COUPONS_SCHEMA = {
  "id": "/Coupons",
  "items": {
    "id": "/items",
    "properties": {
      "Description": {
        "type": "string"
      },
      "Ean": {
        "type": "string"
      },
      "ExpiryDate": {
        "type": "string"
      },
      "Id": {
        "type": "string"
      },
      "Name": {
        "type": "string"
      },
      "StartDate": {
        "type": "string"
      },
      "Type": {
        "type": "string"
      },
      "VoucherValue": {
        "type": "string"
      }
    },
    "type": "object"
  },
  "type": "array"
};

export default COUPONS_SCHEMA;
```

> *Rewards Schema* in rewards.js
```
const REWARDS_SCHEMA = {
    "id": "/Rewards",
    "items": {
        "id": "/items",
        "properties": {
            "PromotionId": {

                "type": "string"
            },
            "Reward Amount": {

                "type": "string"
            },
            "RewardType": {

                "type": "string"
            }
        },
        "type": "object"
    },
    "type": "array"
};

export default REWARDS_SCHEMA;
```

Am referencing above defined schemas in **Discounts Schema**
```

import { Validator } from 'jsonschema';
import Coupons from './coupons';
import Rewards from './rewards';
let validator = new Validator();


const DISCOUNTS_SCHEMA = {
  "id": "/Discounts",
  "properties": {
    "Coupons": {
    "$ref": "/Coupons"
    },
    "PromotionalClubCardPoints": {
      "type": "string"
    },
    "Rewards": {
      "$ref": "/Rewards"
    },
    "StaffDiscount": {
      "type": "string"
    },
    "StandardClubCardPoints": {
      "type": "string"
    },
    "TotalClubCardPoints": {
      "type": "string"
    },
    "TotalCoupons": {
      "type": "string"
    },
    "TotalGiftCards": {
      "type": "string"
    },
    "TotalGreenClubCardPoints": {
      "type": "string"
    },
    "TotalSavings": {
      "type": "string"
    },
    "TotalVouchers": {
      "type": "string"
    }
  },
  "type": "object"
};

validator.addSchema(Coupons,'/Discounts');
validator.addSchema(Rewards,'/Discounts');


export default DISCOUNTS_SCHEMA;
```

and getting the below error 
```
 throw new Error('Schema <'+schema+'> already exists with different definition');
        ^

Error: Schema <[object Object]> already exists with different definition
    at Validator.addSubSchema (/Users/repo/node_modules/jsonschema/lib/validator.js:72:15)
    at Validator.addSubSchemaArray (/Users/repo/node_modules/jsonschema/lib/validator.js:99:10)
    at Validator.addSubSchema (/Users/repo/node_modules/jsonschema/lib/validator.js:80:8)
    at Validator.addSchema (/Users/repo/node_modules/jsonschema/lib/validator.js:48:8)
    at Object.<anonymous> (/Users/repo/src/schema/discounts.js:47:11)
    at Module._compile (module.js:570:32)
    at loader (/Users/repo/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/repo/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
```

**Comment by nbering** [comment]
```
validator.addSchema(Coupons,'/Discounts');
validator.addSchema(Rewards,'/Discounts');
```

Your definitions have the right IDs in them, but you're registering them under the same ID. Try:

```
validator.addSchema(Coupons,'/Coupons');
validator.addSchema(Rewards,'/Rewards');
```

**Comment by SanjeevMurthy** [comment]
Did the changes as suggested, but still am facing the same issue...


I have referencing the same "Rewards" schema in another schema, in "Lineitems" schema. Is this causing issue ??

```
import { Validator } from 'jsonschema';
import Rewards from './Rewards';
let validator = new Validator();


const LINEITEMS_SCHEMA = {
  "id": "/Lineitems",
  "items": {
    "id": "/items",
    "properties": {
      "AddedDateTime": {
        "type": "string"
      },
      "AgerageWeight": {
        "type": "string"
      },
      "BaseProductId": {
        "type": "string"
      },
      "ChargeProductType": {
        "type": "string"
      },
      "ChoiceQuantity": {
        "type": "string"
      },
      "ChoiceWeight": {
        "type": "string"
      },
      "ProductId": {
        "type": "string"
      },
      "QuantityOriginal": {
        "type": "string"
      },
      "Rewards": {
        "$ref": "/Rewards" .   >>> here
      },
      "SubstitutionOption": {
        "type": "string"
      }
    },
    "type": "object"
  },
  "type": "array"
};

validator.addSchema(Rewards,'/Rewards');

export default LINEITEMS_SCHEMA;
```

**Comment by nbering** [comment]
If you were sharing an instance of validator, that would seem to be possible, but it doesn't look like you are.

Is the stack trace exactly the same? Did you rebuild before running again?

**Comment by nbering** [comment]
Oh. I think I see it. 3 of your schemas have a subschema with an id of `/items`, but they are all different. Try changing the ids for the subschemas to something unique.

**Comment by mikbur** [comment]
The same message with relay-compiler 1.5.0 and the schema-file in the source-folder. Downgrading to 1.4.1  solved the problem or moving the schema out of source folder.

**Comment by awwright** [comment]
I'll take a look at this. The error message should be a little bit more descriptive than just `<[Object object]>`.

I would guess, however, the issue is that multiple schemas with the same URI are being defined. In this case, <`/items`>

**Comment by awwright** [comment]
I added a test and fixed the error message in 03ca6827027fcc9bf75afa14cf11220bf9e4e67c.

To avoid problems with importing, use a full URI like `http://example.com/schema.json`.

I can't tell for sure if this solves the problem; if you run into this after the next release, please file a new issue.
